### PR TITLE
xbutil2: fix ps kernel name n instance number

### PR DESCRIPTION
**Pull request by sarab96**
After changes:
...
30      kernel_vcu_decoder_30         0x0             0       (IDLE)
31      kernel_vcu_decoder_31         0x0             0       (IDLE)
32      kernel_vcu_encoder_0          0x0             0       (IDLE)
33      kernel_vcu_encoder_1          0x0             0       (IDLE)

Earlier to changes:
...
30      kernel_vcu_decoder_30     0x0             0       (IDLE)
31      kernel_vcu_decoder_31     0x0             0       (IDLE)
32      kernel_vcu_encoder_32     0x0             0       (IDLE)
33      kernel_vcu_encoder_33     0x0             0       (IDLE)
....

### CODE REVIEW

File: src/runtime_src/core/tools/common/ReportCu.cpp [hunk 1]
```diff
@@ -131,6 +132,22 @@ populate_cus(const xrt_core::device *device)
...
+
+    std::vector<char> buf = xrt_core::device_query<xrt_core::query::ps_kernel>(device);
+    const ps_kernel_node *map = reinterpret_cast<ps_kernel_node *>(buf.data());
+    if(map->pkn_count < 0)
+        return -EINVAL;
+
+    for (unsigned int i = 0; i < map->pkn_count; i++)
+        psKernels.emplace_back(map->pkn_data[i]);
+
+    return 0;
```
Comment (AShivangi) [inline src/runtime_src/core/tools/common/ReportCu.cpp:148]: Indentation. Should be 2 spaces not 4

File: src/runtime_src/core/tools/common/ReportCu.cpp [hunk 2]
```diff
@@ -160,22 +177,40 @@ populate_cus_new(const xrt_core::device *device)
...
-        std::string scu_i = scu_name.substr(found + 3);
-        scu_name = scu_name.substr(0, found - 1);
-        scu_name.append(scu_i);
+    std::string scu_name = "Illegal";
+    if (psk_inst >= psKernels.size()) {
+      scu_name = stat.name; //This means something is wrong
+    } else {
+      scu_name = psKernels.at(psk_inst).pkd_sym_name;
+      scu_name.append("_");
+      scu_name.append(std::to_string(num_scu));
```
Comment (AShivangi) [inline src/runtime_src/core/tools/common/ReportCu.cpp:197]: It's hard to understand what is happening here, can you add a comment to provide an example of the before and after scu_name would look like

File: src/runtime_src/core/tools/common/ReportCu.cpp [hunk 3]
```diff
@@ -160,22 +177,40 @@ populate_cus_new(const xrt_core::device *device)
...
     ptCu.add_child( std::string("status"),	get_cu_status(stat.status));
     pt.push_back(std::make_pair("", ptCu));
+
+    if (psk_inst >= psKernels.size()) {
+      continue;
+    }
+    num_scu++;
+    if (num_scu == psKernels.at(psk_inst).pkd_num_instances) {
+        num_scu = 0;
+        psk_inst++;
```
Comment (AShivangi) [inline src/runtime_src/core/tools/common/ReportCu.cpp:212]: indentation

**Review by AShivangi — CHANGES_REQUESTED** [review]
Minor comments

File: src/runtime_src/core/tools/common/ReportCu.cpp (hunk 2 above)
Comment (sarab96) [reply src/runtime_src/core/tools/common/ReportCu.cpp:199]: Done

File: src/runtime_src/core/tools/common/ReportCu.cpp (hunk 1 above)
Comment (sarab96) [reply src/runtime_src/core/tools/common/ReportCu.cpp]: Done

File: src/runtime_src/core/tools/common/ReportCu.cpp (hunk 3 above)
Comment (sarab96) [reply src/runtime_src/core/tools/common/ReportCu.cpp]: Done

File: src/runtime_src/core/tools/common/ReportCu.cpp [hunk 4]
```diff
@@ -135,17 +135,17 @@ populate_cus(const xrt_core::device *device)
 int 
 getPSKernels(std::vector<ps_kernel_data> &psKernels, const xrt_core::device *device)
 {
-    std::string errmsg;
+  std::string errmsg;
```
Comment (AShivangi) [inline src/runtime_src/core/tools/common/ReportCu.cpp:138]: this isn't used anywhere

**Review by AShivangi — APPROVED** [review]
One minor comment about unused variable, but the rest looks good

**Pull request merged**